src/index.js

```diff
@@ -1,44 +1,28 @@
 import './style.css';
+import displayTasks from './modules/display.js';
+import addTask from './modules/addTask.js';
+import { getTaskListFromLocalStorage, saveTaskListToLocalStorage } from './modules/events.js';
 
-const listArray = [
-  {
-    description: 'Go on holidays',
-    completed: false,
-    index: '1',
-  },
-  {
-    description: 'Wash utensils',
-    completed: false,
-    index: '2',
-  },
-  {
-    description: 'Watch movies',
-    completed: false,
-    index: '3',
-  }, {
-    description: 'Pray to God',
-    completed: false,
-    index: '3',
-  },
-];
+// Function to clear all completed tasks (checkbox checked)
+const clearCompletedTasks = () => {
+  const taskList = getTaskListFromLocalStorage();
 
-const taskcontainer = document.querySelector('.tasklist');
-taskcontainer.innerHTML = '';
+  // Use the filter() method to create a new array containing only the incomplete tasks
+  const incompleteTasks = taskList.filter((task) => !task.checked);
 
-const displayList = () => {
-  listArray.forEach((list) => {
-    const li = document.createElement('li');
-    li.classList.add('newlist');
-    li.innerHTML = `
-    <ul class="list">
-      <li>
-       <input type="checkbox" name="checkbox" id="check" />
-       <label for="label">${list.description}</label
-       ><span class="material-symbols-outlined"> more_vert </span>
-      </li>
-     </ul>`;
-    taskcontainer.appendChild(li);
-  });
+  // Save the updated task list to local storage
+  saveTaskListToLocalStorage(incompleteTasks);
+
+  // Display the updated task list
+  displayTasks(incompleteTasks);
```

Comment:
- [ ] Kindly make sure to update the tasks `index` after clearing all completed tasks. Remember always task `index` should start at 1 (if the task is the fourth its index should always be 4)
```suggestion
 // Updating all task's index
  incompleteTasks.forEach((task, i) => {
    task.index = i + 1;
  });

  // Save the updated task list to local storage
  saveTaskListToLocalStorage(incompleteTasks);

  // Display the updated task list
  displayTasks(incompleteTasks);
```